lib/src/ruby/module.cc

```diff
@@ -112,20 +113,47 @@ extern "C" {
      */
     void LIBFACTER_EXPORT Init_libfacter()
```

Comment:
At this point this function looks almost nothing like the version in stable or in master. It needs proper review